devdoc/docs/ContributeDocs.md

```diff
@@ -0,0 +1,197 @@
+#### Adding or modifying a Language keyword, Standard Library function, or command line tool action
+This needs some changes to existing documentation so the best way to provide the information is in a documentation Jira issue.
+If it s a new keyword, function, or action, a brief overview should be included.
+For a Standard Library function, the developer should update the Javadoc comment in the appropriate ECL file. For a command line tool change, the developer should update the Usage section of the code.
+#### Adding a new feature that requires an overview.
+This is a candidate for either an MD file, a blog, or both. Since there should have been some sort of design specification document, that could easily be repurposed as a good start for this.  
+#### A feature/function that is only used internally to the system
+Since this is information that is probably only of interest to other developers, a write-up in the form of an MD file in the repo is the best approach. If it affects end-users or operations, then a more formal document or a blog might be a good idea. 
+If it affects existing deployments or existing code, then a Red Book notice might also be needed.
+#### Extending the tests in the regression suite
+New tests are frequently added and the regression suite readme files should be updated at the same time. 
+If the tests are noteworthy, we could add a mention in the Platform Release Notes.  
+### Placement
+In general, it makes sense to keep simple documentation near the code. For
+example, a document about ECL Agent should go in the ECLAgent folder.
+However, there are times where that is either not possible or a document
+may cover more than one component. In those cases, there are a few
+options as shown below.
+
+#### Other Folders
+
+**devdoc**: This is a general folder for any developer document.
+
+**devdoc**/**docs**: This is a folder for documents about documentation.
+
+**userdoc**: This is a collection of docs targeted toward the end-user
+rather than developers.
+
+This is primarily for informal documents aimed at end-users. This info can and should be incorporated into the formal docs. 
+**userdoc**/**troubleshoot**: Information related to troubleshooting
+particular components
+
+**userdoc**/**azure**: Useful information about Azure Cloud portal and
+cli
+
+**userdoc**/**roxie**: Useful information for running roxie
+
+**userdoc**/**thor**: Useful information for running thor
+
+**userdoc**/**blogs**: COMING SOON: Location and storage of original
+text for blogs. It also has docs with guidelines and instructions on
+writing Blogs
+
+#### Pull Requests
+
+You can include your documentation with your code in a Pull Request or
+create a separate Jira and Pull Request for the documentation. This
+depends on the size of the code and doc. For a large project or change, a separate Pull request for the documentation is better. 
```

Comment:
need to say why it is better.

a separate Pull request may allow the code change to be merged more quickly?